codeworld-api/src/CodeWorld/Driver.hs

```diff
@@ -49,13 +49,13 @@ import Control.Exception
 import Control.Monad
 import Control.Monad.Trans (liftIO)
 import Data.Char (chr)
-import Data.List (find, zip4)
+import Data.List (find, zip4, intercalate)
 import Data.Maybe (fromMaybe, isNothing, mapMaybe)
 import Data.Monoid
 import Data.Serialize
 import Data.Serialize.Text
 import qualified Data.Text as T
-import Data.Text (Text, pack, singleton)
+import Data.Text (Text, pack, singleton, intercalate)
```

Comment:
I suggest not importing `Data.Text.intercalate`.  Then you can use `intercalate` unqualified on lists.  (If you DO need `Data.Text.intercalate`, you can already use it as `T.intercalate` because of the import on line 57)